### PR TITLE
Feature/boost

### PR DESCRIPTION
**Pull request by gjbex**
This PR contains easyconfigs for the Itnel 2015a toolchain:
* Boost 1.57.0
* GMP 5.1.3
* MPFR 3.1.2
* MPC 1.0.2
* GSL 1.16
* sparsehash 2.0.2
* PCRE 8.36
* SWIG 3.0.3

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/m/MPC/MPC-1.0.2-intel-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+sources = [SOURCELOWER_TAR_GZ]
+
+dependencies = [('GMP', '5.1.3'),
+                ('MPFR', '3.1.2')]
+
+runtest = 'check'
+
+sanity_check_paths = {
+                      'files': ['lib/libmpc.so', 'include/mpc.h'],
+                      'dirs': []
```
Comment (boegel) [inline easybuild/easyconfigs/m/MPC/MPC-1.0.2-intel-2015a.eb]: @gjbex: can you fix these style issues?

```python
sanity_check_paths = {
    'files': ['lib/libmpc.so', 'include/mpc.h'],
    'dirs': [],
}
```

File: easybuild/easyconfigs/m/MPFR/MPFR-3.1.2-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+name = 'MPFR'
+version = '3.1.2'
+
+homepage = 'http://www.mpfr.org'
+description = """The MPFR library is a C library for multiple-precision 
+ floating-point computations with correct rounding."""
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+
+source_urls = ['http://www.mpfr.org/mpfr-%s/' % version]
```
Comment (boegel) [inline easybuild/easyconfigs/m/MPFR/MPFR-3.1.2-intel-2015a.eb]: please change to:

```python
source_urls = ['http://www.mpfr.org/mpfr-%(version)s/']
```

File: easybuild/easyconfigs/m/MPC/MPC-1.0.2-intel-2015a.eb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+easyblock = 'ConfigureMake'
+
+name = 'MPC'
+version = '1.0.2'
+
+homepage = 'http://www.multiprecision.org/'
+description = """Gnu Mpc is a C library for the arithmetic of complex numbers with arbitrarily high precision and correct rounding of the result. It extends the principles of the IEEE-754 standard for fixed precision real floating point numbers to complex numbers, providing well-defined semantics for every operation. At the same time, speed of operation at high precision is a major design goal."""
+
```
Comment (boegel) [inline easybuild/easyconfigs/m/MPC/MPC-1.0.2-intel-2015a.eb]: drop empty line

File: easybuild/easyconfigs/m/MPC/MPC-1.0.2-intel-2015a.eb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+easyblock = 'ConfigureMake'
+
+name = 'MPC'
+version = '1.0.2'
+
+homepage = 'http://www.multiprecision.org/'
+description = """Gnu Mpc is a C library for the arithmetic of complex numbers with arbitrarily high precision and correct rounding of the result. It extends the principles of the IEEE-754 standard for fixed precision real floating point numbers to complex numbers, providing well-defined semantics for every operation. At the same time, speed of operation at high precision is a major design goal."""
```
Comment (boegel) [inline easybuild/easyconfigs/m/MPC/MPC-1.0.2-intel-2015a.eb]: break long description up in separate lines

File: easybuild/easyconfigs/s/sparsehash/sparsehash-2.0.2-intel-2015a.eb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+easyblock = 'ConfigureMake'
+
+name = 'sparsehash'
+version = '2.0.2'
+
+homepage = 'https://code.google.com/p/sparsehash/'
+description = """An extremely memory-efficient hash_map implementation. 2 bits/entry overhead! The SparseHash library contains several hash-map implementations, including implementations that optimize for space or speed."""
```
Comment (boegel) [inline easybuild/easyconfigs/s/sparsehash/sparsehash-2.0.2-intel-2015a.eb]: break up long line (max 120 chars)

**Comment by gjbex** [comment]
Test report by @gjbex
**FAILED**
Build succeeded for 7 out of 8 (8 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/4b79da560630a75c8f37 for a full test report.

**Comment by gjbex** [comment]
All issues should be resolved.

**Comment by gjbex** [comment]
Test report by @gjbex
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.6
See https://gist.github.com/014d0cab920b1bc11f5c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/d57434bca235dec06b6b for a full test report.

**Comment by boegel** [comment]
Looking great now, going in. Thanks @gjbex!

**Pull request merged**